### PR TITLE
python3.pkgs.pysdl2: fix build

### PR DESCRIPTION
**Pull request by povik**
Update patches after version bump.

I ran examples bundled with the library and they seem to run okay.

ZHF: #97479

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
Builds successfully on macos 10.14. Seems like a slightly better approach than previously.

File: pkgs/development/python-modules/pysdl2/default.nix [hunk 1]
```diff
@@ -21,15 +21,15 @@ buildPythonPackage rec {
...
+      "SDL2_ttf-path-from-nix" "${SDL2_ttf}/lib/libSDL2_ttf${stdenv.hostPlatform.extensions.sharedLibrary}"
     substituteInPlace sdl2/sdlimage.py --replace \
-      "DLL(\"SDL2_image\")" "DLL('${SDL2_image}/lib/libSDL2_image${stdenv.hostPlatform.extensions.sharedLibrary}')"
+      "SDL2_image-path-from-nix" "${SDL2_image}/lib/libSDL2_image${stdenv.hostPlatform.extensions.sharedLibrary}"
     substituteInPlace sdl2/sdlgfx.py --replace \
-     "DLL(\"SDL2_gfx\")" "DLL('${SDL2_gfx}/lib/libSDL2_gfx${stdenv.hostPlatform.extensions.sharedLibrary}')"
+      "SDL2_gfx-path-from-nix" "${SDL2_gfx}/lib/libSDL2_gfx${stdenv.hostPlatform.extensions.sharedLibrary}"
     substituteInPlace sdl2/sdlmixer.py --replace \
-     "DLL(\"SDL2_mixer\")" "DLL('${SDL2_mixer}/lib/libSDL2_mixer${stdenv.hostPlatform.extensions.sharedLibrary}')"
+      "SDL2_mixer-path-from-nix" "${SDL2_mixer}/lib/libSDL2_mixer${stdenv.hostPlatform.extensions.sharedLibrary}"
```
[the remark this answers is not in the document]
Comment (jonringer) [reply pkgs/development/python-modules/pysdl2/default.nix:32]: however, i forgot to post this review like 3 days ago. I think what you have works though

[the remark this answers is not in the document]
Comment (povik) [reply pkgs/development/python-modules/pysdl2/default.nix]: I see, that makes for cleaner code. Take a look at the updated commit.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 98300 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>6 packages built:</summary>  <ul>    <li>python27Packages.pysdl2</li>    <li>python27Packages.pytmx</li>    <li>python37Packages.pysdl2</li>    <li>python37Packages.pytmx</li>    <li>python38Packages.pysdl2</li>    <li>python38Packages.pytmx</li>  </ul></details>

**Pull request merged**